### PR TITLE
use torch.cummax in corner_pool for torch 1.5+

### PR DESCRIPTION
**Pull request by tianyuandu**
#3008

### CODE REVIEW

File: mmdet/ops/corner_pool/corner_pool.py [hunk 1]
```diff
@@ -92,10 +93,29 @@ class CornerPool(nn.Module):
...
+        'right': [3, False],
+        'top': [2, True],
+    }
+
     def __init__(self, mode):
         super(CornerPool, self).__init__()
         assert mode in self.pool_functions
-        self.corner_pool = self.pool_functions[mode]
+        if torch.__version__ >= '1.5.0':
+            self.corner_pool = self.cummax_dim_flip[mode]
```
Comment (hellock) [inline mmdet/ops/corner_pool/corner_pool.py:107]: This is quite weird. I suggest saving `mode` and get dim and flip in `forward()`.

File: mmdet/ops/corner_pool/corner_pool.py [hunk 2]
```diff
@@ -92,10 +93,27 @@ class CornerPool(nn.Module):
         'top': TopPoolFunction,
     }
 
+    cummax_dim_flip = {
+        'bottom': [2, False],
+        'left': [3, True],
+        'right': [3, False],
+        'top': [2, True],
```
Comment (hellock) [inline mmdet/ops/corner_pool/corner_pool.py:100]: It makes more sense to use tuple instead of list.

**Comment by hellock** [comment]
This change should also be submitted to mmcv, since we will switch to use all ops from mmcv.

**Comment by hellock** [comment]
It has been merged to mmcv.
